src/libraries/System.Net.WebSockets.Client/tests/CloseTest.cs

```diff
@@ -523,5 +524,42 @@ await Assert.ThrowsAnyAsync<OperationCanceledException>(async () =>
 
             }), new LoopbackServer.Options { WebSocketEndpoint = true });
         }
+
+        // Regression test for https://github.com/dotnet/runtime/issues/80116.
+        [OuterLoop("Uses Task.Delay")]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public async Task CloseHandshake_ExceptionsAreObserved()
+        {
+            await RemoteExecutor.Invoke(static (typeName) =>
+            {
+                CloseTest test = (CloseTest)Activator.CreateInstance(typeof(CloseTest).Assembly.GetType(typeName), new object[] { null });
+                using CancellationTokenSource timeoutCts = new CancellationTokenSource(TimeOutMilliseconds);
+
+                bool unobserved = false;
+                TaskScheduler.UnobservedTaskException += (obj, args) =>
+                {
+                    unobserved = true;
```

Comment:
Nit: consider saving/logging the exception itself and assert that it's null later, that way we can save time later if e.g. this starts capturing unobserved exceptions from the Http stack.